src/main/java/org/jolokia/docker/maven/util/WaitUtil.java

```diff
@@ -176,6 +181,68 @@ public void cleanUp() { }
 
     // ====================================================================================================
 
+    /**
+     * Check whether a given TCP port is available
+     *
+     */
+    public static class TcpPortChecker implements WaitChecker {
+        private static final int TCP_TIMEOUT = 2000;
+
+        private final List<Integer> ports;
+
+        private final List<InetSocketAddress> pending;
+
+        public TcpPortChecker(String host, List<Integer> ports) {
+            this.ports = ports;
+
+            this.pending = new ArrayList<>();
+            for (int port : ports ) {
+                this.pending.add(new InetSocketAddress(host, port));
+            }
+
+        }
+
+        public List<Integer> getPorts() {
+            return ports;
+        }
+
+        public List<InetSocketAddress> getPending() {
+            return pending;
+        }
+
+        @Override
+        public boolean check() {
+            Iterator<InetSocketAddress> iter = pending.iterator();
+
+            while (iter.hasNext()) {
```

Comment:
For a better performance the checks should be done in parallel with threads. This check is perfectly suitable for that since most of the time its blocking.